parquet-avro/src/main/java/org/apache/parquet/avro/ReflectClassValidator.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.avro;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+abstract class ReflectClassValidator {
+
+  abstract void validate(String className);
+
+  static class PackageValidator extends ReflectClassValidator {
+
+    private final List<String> trustedPackagePrefixes = Stream.of(AvroConverters.SERIALIZABLE_PACKAGES)
+        .map(p -> p.endsWith(".") ? p : p + ".")
+        .collect(Collectors.toList());
+    private final boolean trustAllPackages =
+        trustedPackagePrefixes.size() == 1 && "*".equals(trustedPackagePrefixes.get(0));
+    // The primitive "class names" based on Class.isPrimitive()
+    private static final Set<String> PRIMITIVES = new HashSet<>(Arrays.asList(
```

Comment:
Why do we need this?

Reply:
You mean primitives? I'm just trying to mirror the original behavior without actually loading the class (checking the name only).

Reply:
Just wonder what will happen if we remove this.

Reply:
I'm not a expert in Avro either. Not sure if we can hit this with a primitive, but that's how it was originally implemented here and how it is currently implemented in Avro.